pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
Looking at bash args, I think I might prefer the following:

```
// Scriptl is a shell script to pass to the `-c` argument of `/bin/bash` inside the container
Script string
// Command is the command to run, may not be specified with Shell
Command []string
// Args is a list of arguments that are provided to Command or to Shell - for Command they are placed immediately after the command array items, for Shell they are passed to /bin/bash as positional arguments - i.e. `/bin/bash -c "SHELL" -- ARGS0 ARGS1 ARGS2 ..`
Args []string
```

I think this gives us both what we want - the ability to unambiguously pass a chunk of script to the bash command (that is easy to reason about), the ability to pass args to bash (which is easy to reason about), and the ability to override both with a direct command if necessary.